Add notebook for demonstrating how to use the Federated Collection Discovery API

hrodmn opened the pull request:
- demonstrate how to use the Federated Collection Discovery API

@wildintellect if you think it would be better I'm happy to tack this onto the STAC search notebook you mentioned!

wildintellect commented:
@hrodmn let's slap a big BETA in the title, keep it on it's own page, and add it to the technical search table of contents rst.

hrodmn commented:
> @hrodmn let's slap a big BETA in the title, keep it on it's own page, and add it to the technical search table of contents rst.
> 
> Why - the API isn't stable or supported by and standardized clients yet (e.g. pystac_client)

Great, thanks for the reminder on the disclaimer and TOC addition. The last commit addresses those issues!
